lib/Doctrine/Persistence/Reflection/TypedNoDefaultReflectionProperty.php

```diff
@@ -38,7 +38,10 @@ public function setValue($object, $value = null)
                 unset($this->$propertyName);
             };
             $unsetter = $unsetter->bindTo($object, $this->getDeclaringClass()->getName());
-            $unsetter();
+
+            if ($unsetter !== false) {
+                $unsetter();
+            }
```

Comment:
I think an assertion should be used here, instead of silently not calling the `$unsetter` when `bindTo` fails for some reason.